execution/evm/vm.go

```diff
@@ -971,6 +1017,7 @@ func (vm *VM) call(caller acm.Account, callee acm.MutableAccount, code, input []
 					// add account to the tx cache
 					vm.stateWriter.UpdateAccount(acc)
 					ret, callErr = vm.Call(callee, acc, acc.Code(), args, value, &gasLimit)
+					vm.setReturnDataBuffer(ret)
 				}
 			}
 			// In case any calls deeper in the stack (particularly SNatives) has altered either of two accounts to which
```

Comment:
Add `vm.returnData = ret` here in place of removed setters above

We also need to clear the buffer on entry to `CALL*` on line 925 with `vm.returnData = nil` (a lot of the time this won't matter, but catches some edges cases on early return and matches the spec more clearly)